src/Microsoft.VisualStudio.Web.CodeGeneration.Tools/ProjectFileFinder.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.Tools
+{
+    internal class ProjectFileFinder
+    {
+        private string _initialPath;
+        public ProjectFileFinder(string path)
+        {
+            _initialPath = string.IsNullOrEmpty(path)
+                ? Directory.GetCurrentDirectory()
+                : Path.GetFullPath(path);
+
+            Resolve();
+        }
+
+        private void Resolve()
+        {
+            string resolvedPath = _initialPath;
+            var isDirectory = !File.Exists(_initialPath) && Directory.Exists(_initialPath);
+            if (isDirectory)
+            {
+                var csprojFiles = Directory.EnumerateFiles(_initialPath, "*.csproj");
+                var projectJson = Path.Combine(_initialPath, "project.json");
+
+                if ((File.Exists(projectJson) && csprojFiles.Any())
+                    ||csprojFiles.Count() > 1)
+                {
+                    throw new Exception($"Multiple Project files found in the directory {_initialPath}. Please provide full path to the file to use.");
+                }
+
+                if (csprojFiles.Any())
+                {
+                    resolvedPath = csprojFiles.First();
```

Comment:
Yes, right now if there are multiple csproj files, it will throw an exception. 
https://github.com/aspnet/Scaffolding/pull/306/files/515064b4e32914f8dda0eb6cb965e1ed16b2ce39#diff-801050cdcb972a2526c27aa385203e00R32